api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -341,74 +368,85 @@ public void setDataUseAgreementNameOutOfDate(String newGivenName, String newFami
     userDataUseAgreementDao.save(dataUseAgreements);
   }
 
-  public DbUser setDataUseAgreementBypassTime(Long userId, Timestamp bypassTime) {
-    DbUser user = userDao.findUserByUserId(userId);
-    return updateUserWithRetries(
-        (u) -> {
-          u.setDataUseAgreementBypassTime(bypassTime);
-          return u;
-        },
-        user);
-  }
-
-  public DbUser setComplianceTrainingBypassTime(Long userId, Timestamp bypassTime) {
-    DbUser user = userDao.findUserByUserId(userId);
-    return updateUserWithRetries(
-        (u) -> {
-          u.setComplianceTrainingBypassTime(bypassTime);
-          return u;
-        },
-        user);
-  }
-
-  public DbUser setBetaAccessBypassTime(Long userId, Timestamp bypassTime) {
-    DbUser user = userDao.findUserByUserId(userId);
-    return updateUserWithRetries(
-        (u) -> {
-          u.setBetaAccessBypassTime(bypassTime);
+  private DbUser setBypassTimeWithRetries(
+      long userId,
+      Timestamp bypassTime,
+      BypassTimeTargetProperty targetProperty,
+      BiFunction<DbUser, Timestamp, DbUser> setter) {
+    return setBypassTimeWithRetries(
+        userDao.findUserByUserId(userId), bypassTime, targetProperty, setter);
+  }
+
+  public void setDataUseAgreementBypassTime(Long userId, Timestamp bypassTime) {
+    setBypassTimeWithRetries(
```

Comment:
I was pleased with how this came out. I shrunk these methods to one statement each, and added the audit instrumentation with a single parameter.